packages/core/i18n/src/i18n.ts

```diff
@@ -11,11 +11,12 @@ const intlCache = createIntlCache()
 // typed as any since we don't have access to MessageSource here
 let globIntl: any
 
-export const createI18n = <MessageSource extends Record<string, any>>(locale: SupportedLocales, messages: MessageSource, isGlobal: boolean = false): IntlShapeEx<MessageSource> => {
+export const createI18nEx = <MessageSource extends Record<string, any>>(config: IntlConfigEx<MessageSource>, isGlobal: boolean = false): IntlShapeEx<MessageSource> => {
```

Comment:
suggestion: we need to add TS Doc formatted comment above each function here so that users can properly identity which one they want to import.

In my opinion, the two exported functions here are named too similarly. Could we change the name of the new function to something like `createI18nWithConfig` instead? (I think that is what it's actually doing, right?)
```suggestion
/**
 * {Description of this function}
 * @param {IntlConfigEx<MessageSource>} config {Description of config param}
 * @param {boolean} isGlobal {Description of isGlobal param)
 *
*/
export const createI18nEx = <MessageSource extends Record<string, any>>(config: IntlConfigEx<MessageSource>, isGlobal: boolean = false): IntlShapeEx<MessageSource> => {
```